### PR TITLE
CORE:

### PR DESCRIPTION
**Pull request by manolama**
- Fix the by-ref in GBNAI and wrap some more try catch statements.

### CODE REVIEW

File: core/src/main/java/net/opentsdb/query/processor/groupby/GroupByNumericArrayIterator.java [hunk 1]
```diff
@@ -347,7 +348,7 @@ public void run() {
             // AGG
             try {
               final TimeSeries source = sources != null ? sources.get(startIndex) :
-                result.downstreamResult().timeSeries().get(startIndex);
+                result.downstreamResult().timeSeries().get(arraySources[startIndex]);
```
Comment (mailravi3390) [inline core/src/main/java/net/opentsdb/query/processor/groupby/GroupByNumericArrayIterator.java:351]: Is it possible for arraySources to be null ? (I see one of the constructors setting it to null).

File: core/src/main/java/net/opentsdb/query/processor/groupby/GroupByTimeSeries.java [hunk 2]
```diff
@@ -141,7 +141,7 @@ void addSource(final TimeSeries source, final int idx) {
     }
     // TODO - pool
     if (source_refs == null) {
-      source_refs = new int[16];
+      source_refs = new int[8];
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/query/processor/groupby/GroupByTimeSeries.java:144]: it starts with initial size of 8 and then keeps growing?

File: core/src/main/java/net/opentsdb/query/processor/groupby/GroupByNumericArrayIterator.java (hunk 1 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/query/processor/groupby/GroupByNumericArrayIterator.java:351]: Yeah, that's why we have the `sources != null` check.

File: core/src/main/java/net/opentsdb/query/processor/groupby/GroupByTimeSeries.java (hunk 2 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/query/processor/groupby/GroupByTimeSeries.java:144]: Yup, reducing it for the case wherein we have 1 series per group. Still need to pool it eventually.

**Pull request merged**